sound/soc/sof/intel/hda-dsp.c

```diff
@@ -239,10 +239,18 @@ bool hda_dsp_core_is_enabled(struct snd_sof_dev *sdev,
 
 int hda_dsp_enable_core(struct snd_sof_dev *sdev, unsigned int core_mask)
 {
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	const struct sof_intel_dsp_desc *chip = hda->desc;
 	int ret;
 
-	/* return if core is already enabled */
-	if (hda_dsp_core_is_enabled(sdev, core_mask))
+	/*
+	 * restrict core_mask to chip->cores_mask because on some platforms not all cores can be
+	 * powered up by the host.
+	 */
+	core_mask &= chip->cores_mask;
```

Comment:
this is interesting. So, `chip->cores_mask` is not all DSP cores, but only those, power managed by the host? `.cores_num` too? Looks like `.cores_num` isn't used at all. If anything it might be difficult to follow, IMHO we need at least comments in `struct sof_intel_dsp_desc`

Reply:
the number of cores could be useful in future updates to verify that e.g. if we want a pipeline on core 3, well do we actually have a core3?

That said I wonder if we shouldn't use 'physical_cores' and 'power_managed_cores' to explain what this really means. cores_mask looks like number of cores, it really doesn't help understand the nuance about pm differences.

Reply:
yes, would be good to make this a bit clearer